artemis-server/src/main/java/org/apache/activemq/artemis/core/server/transformer/Transformer.java

```diff
@@ -19,10 +19,18 @@
 import java.util.Map;
 
 import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.core.server.ServerMessage;
 
 public interface Transformer {
```

Comment:
Maybe you should change only the old `server.cluster.Transformer` interface and keep this new one pristine?

Reply:
I'm not sure what you mean here Jeff.  Perhaps you could provide an example?  Note that I was trying to keep this change as isolated as possible.  The only thing I have changed is the transformer interface and added a couple extra classes back.

Reply:
The interface that needs to support backwards compatibility is https://github.com/apache/activemq-artemis/blob/master/artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/Transformer.java not this one (this interface is new in Artemis 2.x)
I was suggesting to add default methods to the cluster's interface instead of adding deprecated  method to the new one.

Reply:
@jmesnil  Perhaps I am missing something here.  But, this new interface is what we use internally.  If I was to add these changes to the old interface I would need to check the Transformer Type at runtime and do a cast before calling the appropriate method.  I wanted to avoid this as it has impact on performance.  The way it is at the moment there's only a performance hit when using the 1.x interfaces.

Reply:
@jmesnil Actually, I get what you're saying.  Move all the default methods into the old interface.